lbs/storage.c

```diff
@@ -226,7 +226,7 @@ storage_read(struct storage_state * S, uint64_t blkno, uint8_t * buf)
 	/* Read the block. */
 	if ((s = storage_util_mkpath(S, fs->start)) == NULL)
 		goto err0;
-	if (disk_read(s, (blkno - fs->start) * S->blocklen,
+	if (disk_read(s, (off_t)((blkno - fs->start) * S->blocklen),
```

Comment:
We do have OFF_MAX... on some systems.  If you look at `storage_init` you'll see that we make sure that the blocks in a file fit into OFF_MAX if defined and INT32_MAX otherwise; also for blocks we didn't write ourselves, `blkno - fs->start < fs->len` which is guaranteed to be small enough since we got it from a `stat` call.